contracts/contracts/coordination/InfractionCollector.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import "./Coordinator.sol";
+import "../../threshold/ITACoChildApplication.sol";
+
+contract InfractionCollector is OwnableUpgradeable {
+    Coordinator public coordinator;
+
+    // Reference to the TACoChildApplication contract
+    ITACoChildApplication public tacoChildApplication;
+
+    // Mapping to keep track of reported infractions
+    mapping(uint32 => mapping(address => bool)) public infractions;
+
+    function initialize(
+        Coordinator _coordinator,
+        ITACoChildApplication _tacoChildApplication
+    ) public initializer {
+        __Ownable_init(msg.sender);
+        coordinator = _coordinator;
```

Comment:
Not sure of best practice in solidity, but should we add some `require` statements before setting  anything? (cc @vzotova )
1. Non-zero addresses
```solidity
require(_coordinator != address(0) && _tacoChildApplication != address(0), "Contracts must be specified");
```
2. Check that the values are not already set
```solidity
require(coordinator == address(0) || tacoChildApplication == address(0), "Contracts already set");
```

Reply:
Bump on the RFCs from @derekpierre and @vzotova

Reply:
@vzotova why move them to constructor? i'm never sure of the best patterns here - especially because tacoChild will need to reference InfractionCollector as well

Reply:
when infraction collector will be deployed - we will have taco child and coordinator already working, that's why it's not in taco child constructor, but would be good idea to have it in constructor for collector (it will work as constant which will save gas)

Reply:
fixed in https://github.com/nucypher/nucypher-contracts/pull/267/commits/128ec8e0478197c6acdafb2abdf6c6672427e618

Reply:
small question, I do not see commit 128ec8e0478197c6acdafb2abdf6c6672427e618 here. Am I missing it or it was lost?